### PR TITLE
Add missing re-export items from Next.js

### PR DESCRIPTION
**Pull request by gvasquez11**
Closes: #634 

### What are the changes and their implications?
re-export Error from next/error
re-export AppProps, InferGetStaticPropsType and InferGetServerSidePropsType from next
### Checklist

- [ ] Tests added for changes
- [ ] User facing changes documented

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -34,3 +34,11 @@ export {
...
 
 export {default as dynamic} from 'next/dynamic'
+
+export {Error} from 'next/error'
+
+export {AppProps} from 'next'
+
+export {InferGetStaticPropsType} from 'next'
+
+export {InferGetServerSidePropsType} from 'next'
```
Comment (flybayer) [inline packages/core/src/index.ts:44]: Great, looking good!

One small change that would be good is to move these exports to around line 15 where other items are being exported from `next`. That way `from 'next'` only appears in this file once. Make sense?

Comment (gvasquez11) [reply packages/core/src/index.ts:44]: yeah make sense, got it :) will get this done, thank you!

**Comment by gvasquez11** [comment]
Currently the only error I'm receiving is 

> semantic error TS2300: Duplicate identifier 'AppProps' 

Do you still want me to add that in the export from 'next' as mentioned before, I noticed that on line 21 it's also used but its being exported from 'next/app'

**Comment by flybayer** [comment]
Oh yes, you're right! I overlooked that — AppProps is already being exported, so don't add it a second time :)

**Comment by gvasquez11** [comment]
Awesome! 
One last thing, when trying to re-export 'Error' from 'error/next' I receive:

> semantic error TS2614:  has no exported member 'Error'

I suspect that 'Error' may not be reading from the .next file properly, unless I made a silly typo.

File: packages/core/src/index.ts [hunk 2]
```diff
@@ -34,3 +36,5 @@ export {
 } from 'next/document'
 
 export {default as dynamic} from 'next/dynamic'
+
+export {Error} from 'next/error'
```
Comment (flybayer) [inline packages/core/src/index.ts:40]: This should fix it
```suggestion
export {default as Error} from 'next/error'
```

**Comment by flybayer** [comment]
Looks like something goofed with your editor. Lint is failing because it's missing a new line character at the end.

Adding a new line at the end of the file should fix that

**Comment by flybayer** [comment]
Woohoo!

@all-contributors add @gvasquez11 for code

**Comment by gvasquez11** [comment]
> Woohoo!
> 
> @all-contributors add @gvasquez11 for code

Yay! Finally got it,
Thank you so much for letting me contribute even if it was a small part :)

**Pull request merged**